R/zchunk_L222.en_transformation.R

```diff
@@ -81,248 +87,540 @@ module_energy_L222.en_transformation_DISABLED <- function(command, ...) {
+
+    A22.subsector_logit %>%
+      write_to_all_regions(c("region", "supplysector", "subsector", "logit.year.fillout", "logit.exponent"), GCAM_region_names) ->
+      L222.SubsectorLogit_en
+
+    if (any(!is.na(A22.subsector_shrwt$year))){
+      A22.subsector_shrwt %>%
+        filter(!is.na(year)) %>%
+        write_to_all_regions(c("region", "supplysector", "subsector", "year", "share.weight"), GCAM_region_names) ->
+        L222.SubsectorShrwt_en
+    }
+
+    if (any(!is.na(A22.subsector_shrwt$year.fillout))){
+      A22.subsector_shrwt %>%
+        filter(!is.na(year.fillout)) %>%
+        write_to_all_regions(c("region", "supplysector", "subsector", "year.fillout", "share.weight"), GCAM_region_names) ->
+        L222.SubsectorShrwtFllt_en
+      }
+
+    # L222.SubsectorInterp_en and L222.SubsectorInterpTo_en: Subsector shareweight interpolation of energy transformation sectors
+
+    if (any(is.na(A22.subsector_interp$to.value))){
+    A22.subsector_interp %>%
+      filter(is.na(to.value)) %>%
+      write_to_all_regions(c("region", "supplysector", "subsector", "apply.to", "from.year", "to.year", "interpolation.function"), GCAM_region_names) ->
+      L222.SubsectorInterp_en
+    }
+
+    if (any(!is.na(A22.subsector_interp$to.value))){
+    A22.subsector_interp %>%
+      filter(!is.na(to.value)) %>%
+      write_to_all_regions(c("region", "supplysector", "subsector", "apply.to", "from.year", "to.year", "to.value", "interpolation.function"), GCAM_region_names) ->
+      L222.SubsectorInterpTo_en
+    }
+
+    # 2c. Technology information
+
+    # L222.StubTech_en: Identification of stub technologies of energy transformation
+    # set up filter to keep all non-first gen bio techs in L222.StubTech_en
+    firstgenbio_techs <- c("corn ethanol", "sugarbeet ethanol", "sugar cane ethanol", "biodiesel")
+
+    # create list of regional stub.technologies
+    A22.globaltech_shrwt %>%
+      write_to_all_regions(c("region", "supplysector", "subsector", "technology"), GCAM_region_names) %>%
+      rename(stub.technology = technology) %>%
+      # Drops region x technology combinations that are not applicable, i.e. any first gen bio techs not listed for a region in A_regions
+      filter(!(stub.technology %in% firstgenbio_techs) | paste(region, stub.technology) %in%
+               c(paste(A_regions$region, A_regions$ethanol), paste(A_regions$region, A_regions$biodiesel))) ->
+      L222.StubTech_en
+
+    # L222.GlobalTechInterp_en: Technology shareweight interpolation of energy transformation sectors
+    A22.globaltech_interp %>%
+      set_years() %>%
+      rename(sector.name = supplysector, subsector.name = subsector) ->
+      L222.GlobalTechInterp_en
+
+    # L222.GlobalTechCoef_en: Energy inputs and coefficients of global technologies for energy transformation
+
+    A22.globaltech_coef %>%
+      gather(year, coefficient, -supplysector, -subsector, -technology, -minicam.energy.input) %>%
```

Comment:
`gather(year, coefficient, matches(YEAR_PATTERN))` probably clearer? Same comment applies below.

Reply:
oh, that's useful!